test/units/testsuite-64.sh

```diff
@@ -40,6 +40,57 @@ helper_check_device_symlinks() {(
     done < <(find "${paths[@]}" -type l)
 )}
 
+# Wait for a specific device link to appear
+# Arguments:
+#   $1 - device path
+#   $2 - number of retries (default: 10)
+helper_wait_for_dev() {
+    local dev="${1:?}"
+    local ntries="${2:-10}"
+    local i
+
+    for ((i = 0; i < ntries; i++)); do
+        test ! -e "$dev" || return 0
+        sleep .2
+    done
+
+    return 1
+}
+
+# Wait for the lvm2-pvscan@.service of a specific device to finish
+# Arguments:
+#   $1 - device path
+#   $2 - number of retries (default: 10)
+helper_wait_for_pvscan() {
+    local dev="${1:?}"
+    local ntries="${2:-10}"
+    local MAJOR MINOR pvscan_svc real_dev
+
+    # Sanity check we got a valid block device (or a symlink to it)
+    real_dev="$(readlink -f "$dev")"
+    if [[ ! -b "$real_dev" ]]; then
+        echo >&2 "ERROR: '$dev ($real_dev) is not a valid block device'"
+        return 1
+    fi
+
+    # Get major and minor numbers from the udev database
+    # (udevadm returns MAJOR= and MINOR= expressions, so let's pull them into
+    # the current environment via `source` for easier parsing)
+    source <(udevadm info -q property "$real_dev" | grep -E "(MAJOR|MINOR)=")
```

Comment:
It is little bit tricky, but you can use `udevadm info --export-prefix='' --device-id-of-file /dev/sda`.

Reply:
Note that `--export-prefix=''` is not important, if you modify `$MAJOR` -> `$INFO_MAJOR` and also for `$MINOR`.

Reply:
Oooh, that's way nicer, thanks!

Reply:
Hmpf, I guess this won't work in this scenario, since it returns the MAJOR:MINOR pair of the device the `/dev/sdx` file is on (in this case the `/dev` pseudo-fs), not the block device itself:

```
# udevadm info --export-prefix='' --device-id-of-file /dev/sdi   
MAJOR=0
MINOR=5
# lsblk -rno MAJ:MIN /dev/sdi
8:128
# mountpoint -d /dev
0:5
```

Reply:
Indeed... I misunderstand the option. It just print the underlying device. I cannot understand the motivation of providing that command line option. That's quite useless.

Anyway, sorry for the noise.

Reply:
No worries, I also thought it returns the dev ID of the respective block device.